### ISSUE TITLE
Consider using Functional Options

### ISSUE THREAD

**Issue by caglar10ur** [lxc/go-lxc#24]
See http://dotgo.sourcegraph.com/post/99643162983/dave-cheney-functional-options and https://github.com/pkg/term/blob/master/term.go#L22

**Comment by caglar10ur** [comment]
I still a have strong urge to do this and break the API even after this many year  :) Functional options are (eg; go-grpc style) is really my favorite way. 

Maybe we can do that and bump the bindings to v3...

**Comment by brauner** [comment]
I don't mind so Ack from me. @stgaber, @hallyn?

**Comment by tomponline** [comment]
@caglar10ur the link above is broken now, but I think you're referring to this article https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis 

Sounds good :)

**Comment by caglar10ur** [comment]
@tomponline that is correct. go-grpc also uses that technique and the more I use it over the years the more I started to appreciate it. I don't know how would look like or fit into go-lxc but I'll give it a go in a branch and ping you to see what do you think about it before going ahead and changing everything.
